cumulusci/cli/cci.py

```diff
@@ -1425,83 +1425,235 @@ def task_info(runtime, task_name):
-)
-@click.option(
-    "--no-prompt",
-    is_flag=True,
-    help="Disables all prompts.  Set for non-interactive mode use such as calling from scripts or CI systems",
-)
-@pass_runtime(require_keychain=True)
-def task_run(runtime, task_name, org, o, debug, debug_before, debug_after, no_prompt):
+class RunTaskCommand(click.MultiCommand):
+    def list_commands(self, ctx):
+        tasks = self._get_configured_tasks(RUNTIME)
+        return sorted(tasks)
 
-    # Get necessary configs
-    org, org_config = runtime.get_org(org, fail_if_missing=False)
-    task_config = runtime.project_config.get_task(task_name)
+    def _get_configured_tasks(self, runtime):
+        return list(
+            runtime.project_config.config["tasks"].keys()
+            if runtime.project_config
+            else runtime.global_config.config["tasks"].keys()
+        )
 
-    # Get the class to look up options
-    class_path = task_config.class_path
-    task_class = import_global(class_path)
+    def get_command(self, ctx, task_name):
+        RUNTIME._load_keychain()
+        task_config = RUNTIME.project_config.get_task(task_name)
+        task_class = import_global(task_config.class_path)
+        task_options = task_class.task_options
 
-    # Parse command line options and add to task config
-    if o:
-        if "options" not in task_config.config:
-            task_config.config["options"] = {}
-        for name, value in o:
-            # Validate the option
-            if name not in task_class.task_options:
-                raise click.UsageError(
-                    f'Option "{name}" is not available for task {task_name}'
+        params = self._get_default_command_options()
+        params.extend(self._get_click_options_for_task(task_options))
+
+        def run_task(*args, **kwargs):
+            """Callback function that executes when the command fires."""
+            org, org_config = RUNTIME.get_org(
+                kwargs.pop("org", None), fail_if_missing=False
+            )
+
+            # Overwrite any existing configured options with those
+            # being passed in as long as their value != None.
+            # Values come in as None when they are specified via a cumulusci.yml file
+            for name, value in kwargs.items():
+                if value and name in task_options.keys():
+
+                    # Add options if they aren't present on this task
+                    # they are defined on the parent class
+                    if "options" not in task_config.config:
+                        task_config.config["options"] = {}
+                    task_config.config["options"][name] = value
+
+            try:
+                task = task_class(
+                    task_config.project_config, task_config, org_config=org_config
                 )
 
-            # Override the option in the task config
-            task_config.config["options"][name] = value
+                if kwargs.pop("debug_before", None):
```

Comment:
```suggestion
                if kwargs.get("debug_before"):
```

I don't see a reason to remove it from kwargs